server/src/user/user.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { ForbiddenException, Injectable } from '@nestjs/common';
+
+import { AssertHasRole } from './types';
+import { User } from '../schemas/user.schema';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel(User.name) private readonly userModel: Model<User>) {}
+
+  public async assertHasRole({ userId, role, message = 'Access denied' }: AssertHasRole) {
+    if (typeof userId !== 'string') {
```

Comment:
I don't think that is a possible scenario once you have ID validation on the controller-DTO level